sdks/java/core/src/main/java/org/apache/beam/sdk/options/PipelineOptionsFactory.java

```diff
@@ -873,6 +873,9 @@ private static void terminalPrettyPrint(PrintStream out, String[] words) {
       if (getterMethod != null) {
         Type getterPropertyType = getterMethod.getGenericReturnType();
         Type setterPropertyType = method.getGenericParameterTypes()[0];
+        if (getterPropertyType.equals(ValueProvider.class)) {
```

Comment:
It looks like we skip any type analysis if a VP is set. Why is that? -- Don't we want these checks still?